spec/Observable-spec.ts

```diff
@@ -972,4 +973,239 @@ describe('Observable.lift', () => {
       } }
     );
   });
+
+  describe('As an async iterable', () => {
+    it('should be able to be used with for-await-of', async () => {
+      const source = new Observable<number>((subscriber) => {
+        subscriber.next(1);
+        subscriber.next(2);
+        subscriber.next(3);
+        subscriber.complete();
+      });
+
+      const results: number[] = [];
+      for await (const value of source) {
+        results.push(value);
+      }
+
+      expect(results).to.deep.equal([1, 2, 3]);
+    });
+
+    it('should unsubscribe if the for-await-of loop is broken', async () => {
+      const source = new Observable<number>((subscriber) => {
+        subscriber.next(1);
+        subscriber.next(2);
+        subscriber.next(3);
+        subscriber.complete();
+      });
+
+      const results: number[] = [];
+      for await (const value of source) {
+        results.push(value);
+        break;
+      }
+
+      expect(results).to.deep.equal([1]);
+    });
```

Comment:
Either a test name is misleading here or the unsubscription is not tested here. Are we expecting the subscription to be unsubscribed earlier if we break out of the loop?

Based on the test name I would imagine something like this:
```ts
    it('should unsubscribe if the for-await-of loop is broken', async () => {
      let state = 'idle';
      const source = new Observable<number>((subscriber) => {
        subscriber.next(1);
        setTimeout(() => subscriber.next(2), 100);
        setTimeout(() => subscriber.next(3), 200);
        setTimeout(() => subscriber.complete(), 500);

        return () => {
          state = 'unsubscribed'
        }
      });

      expect(state).to.equal('idle')

      const results: number[] = [];
      for await (const value of source) {
        results.push(value);
        break;
      }

      expect(state).to.equal('unsubscribed')
      expect(results).to.deep.equal([1]);
    });
```

Reply:
I'll change this one around to test an active subscription count.